source/common/http/http1/codec_impl.cc

```diff
@@ -259,7 +261,13 @@ const ToLowerTable& ConnectionImpl::toLowerTable() {
 }
 
 ConnectionImpl::ConnectionImpl(Network::Connection& connection, http_parser_type type)
-    : connection_(connection) {
+    : connection_(connection), output_buffer_(Buffer::InstancePtr{new Buffer::OwnedImpl()},
+                                              [&]() -> void { this->onBelowLowWatermark(); },
+                                              [&]() -> void { this->onAboveHighWatermark(); }) {
+  uint32_t buffer_limit = connection.bufferLimit();
+  if (buffer_limit > 0) {
+    output_buffer_.setWatermarks(buffer_limit / 2, buffer_limit);
```

Comment:
I just looked and in connection_impl we are doing buffer_limit +1, but in both codecs we are just doing buffer_limit. Any reason for that? Also, I think I commented on this before but it might have gotten lost: I feel like we might want a utility function that we use in all three places just to keep the math consistent? Or at least in both codecs?